python/external_models/hf_runner.py

```diff
@@ -0,0 +1,102 @@
+
+import torch
+import numpy as np
+from loguru import logger
+from typing import List, Tuple
+from abc import ABC, abstractmethod
+from transformers import (
+    AutoModelForCausalLM,
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer,
+    AutoModelForTextEncoding,
+)
+import os
+import numpy as np
+
+import openai
+from openai import OpenAI
+from .external_parser import *
+
+
+class HFTacticGenerator(Generator, Transformer):
+    def __init__(
+        self,
+        **args
+    ) -> None:
+        self.name = args['model']
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            self.name, trust_remote_code=True)
+        device = args['device']
+        if device == "auto":
+            device = get_cuda_if_available()
+        else:
+            device = torch.device(device)
+        logger.info(f"Loading {self.name} on {device}")
+        self.model = AutoModelForCausalLM.from_pretrained(
+            self.name, trust_remote_code=True).to(device)
+
+        self.generation_args: dict[str | str] = {
+            "do_sample": args["do_sample"],
+            "temperature": args['temperature'],  # chat default is 0.8
+            "max_new_tokens": args['max_new_tokens'],
+            "top_p": args['top_p'],  # chat default is 0.8
+            # "length_penalty": args["length_penalty"],
+            "num_return_sequences": args['num_return_sequences'],
+            # "num_beams": self.num_return_sequences,
+            # Here if we use beam search for llms the output are not diverse(few tactics are provided).
```

Comment:
Are the lines of code currently commented out dead or would they be useful in some cases? In general if they are just dead code we do not need to keep them in comments. We can just delete them. Otherwise if they are somewhat optional, we should make them as an option instead of leaving them commented out here. Same for every other commented code.

Reply:
If we activate "num_beams", then all the settings will need to change, so I leave a comment here to see if other needs.

Reply:
I see, in this case we can keep this line. How about other lines of code that are commented out? Can we make sure that all the commented-out code left here has a necessary reason and that we leave a note of that reason alongside the code? Otherwise we should just get rid of unnecessary ones.

Reply:
e.g. lines 42, 68, 71, etc.